infra/README.md

```diff
@@ -0,0 +1,75 @@
+    - `dev`
+    - `staging`
+    - `prod`
+
+### 📂 Directory structure
+
+The infrastructure for this project has extended the Nava platform infrastructure template's [module architecture](../docs/infra/module-architecture.md). It currently looks like this:
+
+```text
+infra/                  Infrastructure code
+  accounts/             Root module for IaC and IAM resources
+  [app_name]/           Application directory: infrastructure for the main application
+  modules/              Reusable child modules
+  networks/             Account level network config (shared across all apps, environments, and terraform workspaces)
+```
+
+Each application directory contains the following:
+
+```text
+  app-config/         Application-level configuration for the application resources (different config for different environments)
+  build-repository/   Docker image repository for the application (shared across environments and terraform workspaces)
+  database/           Configuration for database (different config for different environments)
+  service/            Configuration for containers (load balancer, application service) (different config for different environments)
+```
+
+### 🏗️ Project architecture
+
+#### 🥞 Infrastructure layers
+
+The infrastructure template is designed to operate on different layers.
+
+- Account layer
+- Network layer
+- Build repository layer (per application)
+- Database layer (per application)
+- Service layer (per application)
+
+### 🔀 Project workflow
+
+This project relies on Make targets in the [root Makefile](../Makefile), which in turn call shell scripts in [./bin](../bin). The shell scripts call terraform commands. Many of the shell scripts are also called by the [Github Actions CI/CD](../.github/workflows).
+
+Backend configuration is saved as [`.tfbackend`](https://developer.hashicorp.com/terraform/language/settings/backends/configuration#file) files. Most `.tfbackend` files are named after the environment. For example, the `[app_name]/service` infrastructure resources for the `dev` environment are configured via `dev.s3.tfbackend`. Resources for a module that are shared across environments, such as the build-repository, use `shared.s3.tfbackend`. Resources that are shared across the entire account (e.g. /infra/accounts) use `<account name>.<account id>.s3.tfbackend`.
+
+Generally you should use the Make targets or the underlying bin scripts, but if you want to directly use the `terraform` cli, you need to pass in the `.tfbackend` file to init commands and the `.tfvars` file to other commands:
+
+```sh
+infra/portal/service$ terraform init -backend-config=dev.s3.tfbackend
+infra/portal/service$ terraform apply -var="environment_name=dev"
+```
+
+## 💻 Development
+
+### 1️⃣ First time initialization
+
+To set up this project for the first time (aka it has never been deployed to the target AWS account):
+
+1. Follow the steps as outlined in the [infrastructure documentation README](../docs/infra/README.md) for:
```

Comment:
It's not super confusing, but slightly confusing having this other nested readme file. I'm wondering if it should be combined with this one.

Reply:
Yeah. I can see that. Would it be less confusing if I moved these proposed changes to `/docs/infra/README.md` and dropped this section?

Reply:
Nah, I strongly prefer the project directory including a readme. I'm on a project that initially did the nested docs structure and it made onboarding pretty confusing since setup instructions weren't easily discoverable.

Reply:
Agree. To clarify, I meant specifically: would it be less confusing if I dropped the `## 1️⃣ First time initialization` section or kept the section but just said "look in this other doc"?

Reply:
ohh my bad. If we keep the other doc, I lean slightly towards just pointing to it, otherwise there's some duplicate docs to maintain.

Reply:
Ok cool. I'll update this PR to move some of this content around. Thanks for the review!

Reply:
> Nah, I strongly prefer the project directory including a readme

if i understand correctly, does that mean moving docs from `/docs/infra` to `/infra`, docs from `/docs/app` to `/app`?

if so, i'm open to it and can see the point about discoverability, but can you help me think through how we'd organize docs that don't mirror the code structure? for example, if we're talking about how app integrates with infra, where would that go?

Reply:
oh i get it. that works for me.